packages/grid/_modules_/grid/hooks/features/useColumnResize.tsx

```diff
@@ -1,204 +1,130 @@
-        newWidth = newWidth > MIN_COL_WIDTH ? newWidth : MIN_COL_WIDTH;
-        updateWidth(newWidth);
-      }
-    },
-    [updateWidth],
-  );
+    colDefRef.current = colDef;
+    colElementRef.current = columnsRef.current!.querySelector(
+      `[data-field="${colDef.field}"]`,
+    ) as HTMLDivElement;
 
-  // This a hack due to the limitation of react as I cannot put columnsRef in the dependency array of the effect adding the Event listener
-  const columnsRefState = useStateRef(columnsRef);
+    colCellElementsRef.current = findCellElementsFromCol(colElementRef.current) as NodeListOf<
+      Element
+    >;
 
-  React.useEffect(() => {
-    if (columnsRef && columnsRef.current) {
-      logger.info('Adding resizing event listener');
-      const columnsRefEvents = columnsRef.current;
-      columnsRef.current.addEventListener('mouseup', stopResize);
-      columnsRef.current.addEventListener('mouseleave', handleMouseLeave);
-      columnsRef.current.addEventListener('mouseenter', handleMouseEnter);
-      columnsRef.current.addEventListener('mousemove', handleMouseMove);
-
-      return () => {
-        columnsRefEvents.removeEventListener('mouseup', stopResize);
-        columnsRefEvents.removeEventListener('mouseleave', handleMouseLeave);
-        columnsRefEvents.removeEventListener('mouseenter', handleMouseEnter);
-        columnsRefEvents.removeEventListener('mousemove', handleMouseMove);
-      };
-    }
-    return undefined;
-  }, [
-    columnsRefState,
-    columnsRef,
-    handleMouseLeave,
-    handleMouseDown,
-    handleMouseMove,
-    handleMouseEnter,
-    logger,
-    stopResize,
-  ]);
+    const doc = ownerDocument(apiRef.current.rootElementRef!.current as HTMLElement);
+    doc.body.style.cursor = 'col-resize';
+
+    initialOffset.current =
+      (colDefRef.current.width as number) -
+      (event.clientX - colElementRef.current!.getBoundingClientRect().left);
+
+    // @ts-expect-error fixed in v5
+    doc.addEventListener('mousemove', handleResizeMouseMove);
+    // @ts-expect-error fixed in v5
+    doc.addEventListener('mouseup', handleResizeMouseUp);
+  });
+
+  const stopListening = React.useCallback(() => {
+    const doc = ownerDocument(apiRef.current.rootElementRef!.current as HTMLElement);
+    doc.body.style.removeProperty('cursor');
+    // @ts-expect-error fixed in v5
+    doc.removeEventListener('mousemove', handleResizeMouseMove);
+    // @ts-expect-error fixed in v5
+    doc.removeEventListener('mouseup', handleResizeMouseUp);
+  }, [apiRef, handleResizeMouseMove, handleResizeMouseUp]);
 
   React.useEffect(() => {
     return () => {
-      clearTimeout(mouseLeftTimeout.current);
       clearTimeout(stopResizeEventTimeout.current);
+      stopListening();
     };
-  }, []);
+  }, [stopListening]);
 
-  return handleMouseDown;
+  return React.useMemo(
+    () => ({
+      onMouseDown: handleMouseDown,
+    }),
+    [handleMouseDown],
+  );
```

Comment:
It should be useCallback as it's a function
Also why is the formatting like that? So many rows...

Reply:
useCallback won't work, we expect to return an object.